File: Java/UberVan.java
import java.util.ArrayList;
import java.util.Map;

class UberVan extends Car {
    Map<String, Map<String, Integer>> typeCarAccepted;
    ArrayList<String> seatsMaterial;
    private Integer passenger;
    
    public UberVan(
            String license,
            Account driver,
            String brand, String model,
            Map<String, Map<String, Integer>> typeCarAccepted,
            ArrayList<String> seatsMaterial) {
        super(license, driver);
        this.typeCarAccepted = typeCarAccepted;
        this.seatsMaterial = seatsMaterial;
    }

    @Override
    public void setPassenger(Integer passenger){
        if(passenger == 6){
            this.passenger = passenger;
        }else{
            System.out.println("Necesitas asignar 6 pasajeros");
        }
        
    }
}

File: JS/UberVan.js
import Car from './Car.js';

class UberVan extends Car{
    constructor(id, license, driver, passengers, typeCarAccepted, seatsMaterial){
        super(id, license, driver, passengers);
        this.typeCarAccepted = typeCarAccepted;
        this.seatsMaterial = seatsMaterial;
    }
    showDataCar(){
        console.log(`
            ID: ${toyota.id}
            License: ${toyota.license}
            Driver Name: ${toyota.driver.name}
            Driver Document: ${toyota.driver.document}
            Passengers: ${toyota.passengers}
            Accepted brands: ${toyota.typeCarAccepted}
            Seats Material: ${toyota.seatsMaterial}
        `);
    }
}


const toyota = new UberVan('23123', 'fksdk23', {name: "Felipe Moreno", document: 1036401096}, 6, ['toyota', 'ford'], 'leather');

toyota.showDataCar();
